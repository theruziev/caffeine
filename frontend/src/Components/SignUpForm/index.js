import React from 'react'

import useForm from 'react-hook-form'

export default function SignUpForm (props) {
  const { register, handleSubmit, errors, setError } = useForm()
  const onSubmit = data => {
    if (data.password !== data.repeat_password) {
      setError('repeat_password', 'notMatch', 'Password and repeat password is not match.')
    }
  }

  return (

    <form onSubmit={handleSubmit(onSubmit)}>

      <div className='field'>
        <label className='label'>Email</label>
        <div className='control'>
          <input ref={register({ required: true })} className={`input ${errors.username && 'is-danger'}`} name='username' type='text' placeholder='cafe@ephiopia.com' />
        </div>
        {errors.username && <p className='help is-danger'>Username or password is required</p>}
      </div>

      <div className='field'>
        <label className='label'>Password</label>
        <div className='control'>
          <input ref={register({ required: true })} className={`input ${errors.password && 'is-danger'}`} name='password' type='password' placeholder='You password' />
        </div>
        {errors.password && <p className='help is-danger'>The password is required</p>}

      </div>

      <div className='field'>
        <label className='label'>Repeat Password</label>
        <div className='control'>
          <input
            ref={register({ required: true })} className={`input ${errors.repeat_password && 'is-danger'}`}
            name='repeat_password' type='password' placeholder='Repeat you password'
          />
        </div>
        {errors.repeat_password && <p className='help is-danger'>{errors.repeat_password.message}</p>}

      </div>

      <div className='field  is-grouped is-grouped-right'>
        <div className='control'>
          <button className='button is-primary' type='submit'>Sign Up</button>
        </div>
      </div>
    </form>
  )
}
