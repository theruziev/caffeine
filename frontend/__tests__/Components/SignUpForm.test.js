
import React from "react";
import renderer from 'react-test-renderer';
import SignUpForm from '../../src/Components/SignUpForm'
import { shallow } from 'enzyme'



describe('<SignUpForm />', function () {

  it('snapshot renders', () => {
    const component = renderer.create(<SignUpForm/>);
    let componentJSON = component.toJSON();
    expect(componentJSON).toMatchSnapshot();
  });

  it('render 2 input', () => {
    const component = shallow(<SignUpForm/>);
    expect(component.find('input').length).toEqual(3);
  });


})
