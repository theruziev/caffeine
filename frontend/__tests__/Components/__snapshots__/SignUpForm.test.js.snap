// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> snapshot renders 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label
      className="label"
    >
      Email
    </label>
    <div
      className="control"
    >
      <input
        className="input undefined"
        name="username"
        placeholder="cafe@ephiopia.com"
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      Password
    </label>
    <div
      className="control"
    >
      <input
        className="input undefined"
        name="password"
        placeholder="You password"
        type="password"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label"
    >
      Repeat Password
    </label>
    <div
      className="control"
    >
      <input
        className="input undefined"
        name="repeat_password"
        placeholder="Repeat you password"
        type="password"
      />
    </div>
  </div>
  <div
    className="field  is-grouped is-grouped-right"
  >
    <div
      className="control"
    >
      <button
        className="button is-primary"
        type="submit"
      >
        Sign Up
      </button>
    </div>
  </div>
</form>
`;
